google/cloud/bigtable/internal/async_row_sampler_test.cc

```diff
@@ -379,6 +379,8 @@ TEST_F(AsyncSampleRowKeysTest, TimerError) {
   // return "call cancelled" than to pass along the exact error.
   EXPECT_THAT(sor,
               StatusIs(StatusCode::kCancelled, HasSubstr("call cancelled")));
+  EXPECT_THAT(sor.status().error_info().metadata(),
```

Comment:
Alternatively, we could add an ErrorInfoMatcher overload to this. I am not sure exactly what it would look like, so I am happy to take this PR as is.

https://github.com/googleapis/google-cloud-cpp/blob/49061d87a865627cae1ecbaef763b7d7a075d6d9/google/cloud/testing_util/status_matchers.h#L174